File: python/merge/Merger_pdf_text.py
from aspose.pdf import (
    Document
)

from aspose.pdf import (
    Document
)

from aspose.pdf.text import (
    TextAbsorber
)


def pdf_to_text():
    path_source1 = "../../TestData/test.pdf"
    path_source2 = "../../TestData/Second/test.pdf"

    # read pdf file to Aspose Document
    first_doc = Document(path_source1)
    second_doc = Document(path_source2)

    # create empty pdf document
    output_doc = Document()

    # set less memory usage with unload instead of fast performance
    output_doc.enable_object_unload = True

    for page in first_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    for page in second_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    # create text absorber for extract text
    text_absorber = TextAbsorber()
    output_doc.pages.accept(text_absorber)
    extracted_text = text_absorber.text

    # save content to text file
    with open("Merger_pdf_text.txt", "w") as f:
        f.write(extracted_text)

File: cpp/merge/Merger_pdf_text.cpp
#include <fstream>
#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/Text/TextAbsorber.h"

using namespace std;
using namespace System;
using namespace Aspose::Pdf;

void pdf_to_text()
{
	String pathSource1 = u"../../TestData/test.pdf";
	String pathSource2 = u"../../TestData/Second/test.pdf";

	// read pdf file to Aspose Document
	System::SharedPtr<Document> firstDoc = MakeObject<Document>(pathSource1);
	auto secondDoc = MakeObject<Document>(pathSource2);

	// create empty pdf document
	auto outputDoc = MakeObject<Document>();

	// set less memory usage with unload instead of fast performance
	outputDoc->set_EnableObjectUnload(true);

	for (int i = 1; i < firstDoc->get_Pages()->get_Count(); i++)
	{
		auto page = firstDoc->get_Pages()->idx_get(i);
		// add page from one document to another directly
		outputDoc->get_Pages()->CopyPage(page);
	}

	for (int i = 1; i < secondDoc->get_Pages()->get_Count(); i++)
	{
		auto page = secondDoc->get_Pages()->idx_get(i);
		// add page from one document to another directly
		outputDoc->get_Pages()->CopyPage(page);
	}

	// create text absorber for extract text
	auto textAbsorber = MakeObject<Aspose::Pdf::Text::TextAbsorber>();
	outputDoc->get_Pages()->Accept(textAbsorber);
	auto extractedText = textAbsorber->get_Text();

	fstream file;
	file.open("text.txt", ios::out);
	file << extractedText;
	file.close();
}
